packages/core/src/data/sources/lazy/axisMeasureSource.js

```diff
@@ -0,0 +1,154 @@
+        ]);
+        this.measureLabeledValues = Object.fromEntries(this.measureEntries);
+    }
+
+    /**
+     * @param {import("../../../spec/data.js").AxisMeasureData} params
+     * @param {import("../../../view/view.js").default} view
+     */
+    constructor(params, view) {
+        /** @type {import("../../../spec/data.js").AxisMeasureData} params */
+        const paramsWithDefaults = {
+            channel: "x",
+            minMeasureSize: 50,
+            hideMeasureThreshold: 10,
+            multiplierValue: 1,
+            alignMeasure: "center",
+            ...params,
+        };
+
+        super(view, paramsWithDefaults.channel);
+        this.params = paramsWithDefaults;
+        this.updateMeasureLabeledValues();
+    }
+    /** Function that calculates the domain coordinates
+     * of the measure, given the alignment parameter,
+     * the current scale domain and the size of the
+     * measure in domain units
+     *
+     * @param {string} alignMeasure
+     * @param {ScaleWithProps} scale
+     * @param {number} measureDomainSize
+     */
+    calculateCoord(alignMeasure, scale, measureDomainSize) {
+        let coordLeft, coordCenter, coordRight;
+        switch (alignMeasure) {
+            case "left":
+            case "bottom": // Fall-through case for 'bottom' along with 'left'
+                coordLeft = scale.domain()[0];
+                coordRight = coordLeft + measureDomainSize;
+                coordCenter = Math.floor((coordLeft + coordRight) / 2);
+                break;
+            case "center":
+                // @ts-ignore
+                coordCenter = scale.invert(0.5);
+                coordLeft = coordCenter - Math.floor(measureDomainSize / 2);
+                coordRight = coordCenter + Math.floor(measureDomainSize / 2);
+                break;
+            case "right":
+            case "top": // Fall-through case for 'top' along with 'right'
+                coordRight = scale.domain()[1];
+                coordLeft = coordRight - measureDomainSize;
+                coordCenter = Math.floor((coordLeft + coordRight) / 2);
+                break;
+            default:
+                console.error("Invalid alignMeasure value");
+                return null; // Return null or handle as necessary for invalid inputs
+        }
+        return [coordLeft, coordCenter, coordRight];
+    }
+
+    onDomainChanged() {
+        /** @type {ScaleWithProps} scale */
+        const scale = this.scaleResolution.scale;
+        const axisLength = this.getAxisLength();
+        const minMeasureSize = this.params.minMeasureSize;
+        const alignMeasure = this.params.alignMeasure;
+        const hideMeasureThreshold = this.params.hideMeasureThreshold;
+        // Find the size (in domain units) of a measure
+        // that shows minMeasureSize pixels wide
+        const minValueSize = Math.floor(
+            // @ts-ignore
+            scale.invert(minMeasureSize / axisLength) - scale.invert(0)
```

Comment:
I tried to eliminate the warning for scale not having the invert attribute (although it definitely has it) by importing the types VegaScale and ScaleWithProps and declaring scale as being of ScaleWithProps type, but it still doesn't work, so I had to use the `// @ts-ignore` comment to allow it to build.

Reply:
Yeh, typings of d3 scales are pain. It's pretty much impossible to get them right.